proto/osmosis/protorev/v1beta1/protorev.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+package osmosis.protorev.v1beta1;
+
+import "cosmos_proto/cosmos.proto";
+import "gogoproto/gogo.proto";
+
+import "cosmos/base/v1beta1/coin.proto";
+
+option go_package = "github.com/osmosis-labs/osmosis/v13/x/protorev/types";
+
+// TokenPairArbRoutes tracks all of the hot routes for a given pair of tokens
+message TokenPairArbRoutes {
+  option (gogoproto.equal) = true;
+
+  // Stores all of the possible hot paths for a given pair of tokens
+  repeated Route arb_routes = 1;
+  // Token denomination of the first asset
+  string token_a = 2;
+  // Token denomination of the second asset
+  string token_b = 3;
+}
+
+// Route is a hot route for a given pair of tokens
+message Route {
+  option (gogoproto.equal) = true;
+
+  // The pool IDs that are travered in the directed cyclic graph (traversed left
+  // -> right)
+  repeated Trade trades = 1;
+}
+
+// Trade is a single trade in a route
+message Trade {
+  option (gogoproto.equal) = true;
+
+  // The pool IDs that are travered in the directed cyclic graph (traversed left
+  // -> right)
+  uint64 pool = 1;
+  // The denom of token A that is traded
+  string denomA = 2;
+  // The denom of token B that is traded
+  string denomB = 3;
+}
+
+// PoolStatistics tracks the arbitrarge statistics for a given pool
+message PoolStatistics {
+  // Mapping between denom and the profits earned for this pool
+  map<string, cosmos.base.v1beta1.Coin> profits = 1;
```

Comment:
Need to be careful that this is deterministic under current SDK serialization :(

https://github.com/cosmos/cosmos-sdk/issues/12816

Reply:
Thanks for highlighting this. 

We'll follow the same standards as you use in the rest of the code base. Can you point us to a canonical example where you're taking data stored in a map and serializing it safely? 

Is the best way to just order by keys and store as a pair of arrays?

Reply:
We don't actually use proto maps anywhere in the codebase at the moment. We annoyingly always do some manual "unrolling" of this, of making a data structure with both items as fields, and make a repeated list of them where we want to include them. We then make a linear time `Get` function. (Asymptotically were still at linear time, because deserializing the map requires reading all of its fields as well)

Super silly that this is the status quo right now :/

Reply:
That makes sense. 

I mean could you point us to a place where you can do this unrolling, so we can adopt this approach. 

Sounds like you don't unpack the whole list of key, value pairs into an in-memory map if I'm understanding correctly? You just iterate over the list each time you need a key. This seems fine (same asymptotic time) if you only access 1 key in expectation each time you unserialize. 

But if we access potentially many in expectation, it feels like it would make more sense to: 
- store using the key value approach you all use 
- iterate over the entire list to store in an in-memory map o(n) 
- do all our logic with o(1) accesses in memory instead of o(n) accesses 
- if there are changes, recreate the list of key, value pairs with a deterministic ordering when we need to store again 

(Disclaimer: I haven't done SDK development, so maybe there's not a way to have this separation of stored data and in memory usage of it as I'm imagining it)

Reply:
Thats good to know! We can do the manual unrolling (as mentioned) and have everything handled in the KV stores by adding a new (key, value) as oppose to using the map. Given that, I'll deprecate PoolStatistics.